jme3-lwjgl3/src/main/java/com/jme3/input/lwjgl/GlfwMouseInput.java

```diff
@@ -69,20 +68,24 @@
 
     private static final int WHEEL_SCALE = 120;
 
-    private LwjglWindow context;
+    private final LwjglWindow context;
     private RawInputListener listener;
     private boolean cursorVisible = true;
+    private Long[] currentCursor;
+    private IntBuffer currentCursorDelays;
+    private int currentCursorFrame = 0;
+    private double currentCursorFrameStartTime = 0.0;
     private int mouseX;
     private int mouseY;
     private int mouseWheel;
     private boolean initialized;
     private GLFWCursorPosCallback cursorPosCallback;
     private GLFWScrollCallback scrollCallback;
     private GLFWMouseButtonCallback mouseButtonCallback;
-    private Queue<MouseMotionEvent> mouseMotionEvents = new LinkedList<MouseMotionEvent>();
-    private Queue<MouseButtonEvent> mouseButtonEvents = new LinkedList<MouseButtonEvent>();
+    private final Queue<MouseMotionEvent> mouseMotionEvents = new ArrayDeque<MouseMotionEvent>();
+    private final Queue<MouseButtonEvent> mouseButtonEvents = new ArrayDeque<MouseButtonEvent>();
```

Comment:
It's ok, but I think we can use the construction ArrayDeque<>() instead of ArrayDeque<MouseButtonEvent>() ;)

Reply:
Sure! Diamonds for the win :)